app/components/enterprise_edition/plan_for_feature.rb

```diff
@@ -37,15 +37,29 @@ module PlanForFeature
       attr_accessor :i18n_scope
     end
 
+    def teaser?
+      feature_key == :teaser
+    end
+
+    def token
+      @token ||= EnterpriseToken.active_trial_tokens.last
+    end
+
     def title
       I18n.t(:title, scope: i18n_scope, default: default_title)
     end
 
     def default_title
-      I18n.t(feature_key, scope: :"ee.features")
+      teaser? ? teaser_title : I18n.t(feature_key, scope: :"ee.features")
+    end
+
+    def teaser_title
+      I18n.t("ee.teaser.title", count: token.days_left || 365, trial_plan: token.plan)
```

Comment:
same, if `EnterpriseEdition::TrialTeaserComponent` was a thing, there would be no `if` but only redefinitions of `default_title`, `description`, and `plan_text`.